docs/src/pages/components/data-grid/filtering/CustomFilterPanelContent.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { DataGridPro, GridLinkOperator, GridToolbar } from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VISIBLE_FIELDS = ['name', 'rating', 'country', 'dateCreated', 'isAdmin'];
+
+const initialState = {
+  filter: {
+    filterModel: {
+      items: [
+        {
+          id: 1,
+          columnField: 'name',
+          operatorValue: 'contains',
+          value: 'D',
+        },
+        {
+          id: 2,
+          columnField: 'name',
+          operatorValue: 'contains',
+          value: 'D',
+        },
+        {
+          id: 3,
+          columnField: 'rating',
+          operatorValue: '>',
+          value: '0',
+        },
+      ],
+    },
+  },
+};
+
+export default function CustomFilterPanelContent() {
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    visibleFields: VISIBLE_FIELDS,
+    rowLength: 100,
+  });
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGridPro
+        {...data}
+        components={{
+          Toolbar: GridToolbar,
+          // Use custom FilterPanel only for deep modification
+          // FilterPanel: MyCustomFilterPanel,
+        }}
+        componentsProps={{
+          filterPanel: {
+            // Force to use "And" operator
+            linkOperators: [GridLinkOperator.And],
+            // Display columns by ascending alphabetical order
+            columnsSort: 'asc',
+            filterFormProps: {
+              // Customize inputs by passing props
+              columnInputProps: {
+                variant: 'outlined',
```

Comment:
There's something wrong using this variant. The input border should be passing behind the label. See https://mui.com/components/text-fields/#select

![image](https://user-images.githubusercontent.com/42154031/151282217-11aed43d-bf68-42a5-b146-41b0ca97d6c4.png)

Reply:
That's because of the `sx: { justifyContent: 'flex-end' }` applied to the filter form which impact the select, but not the label
I have to add the `label` prop to `<Select />` otherwise the outline is not removed under the label